src/hooks/useZkVerify.ts

```diff
@@ -0,0 +1,47 @@
+export function useZkVerify() {
```

Comment:
I would consider this a utility function than a hook bc it doesn't use any of the React features like `useState` or `useEffect`

Reply:
Latest push now utilises useState and useEffect.